### PR TITLE
hard float detection

### PR DESCRIPTION
**Pull request by moonbuggy**
spdMerlin started having problems on my RT-AC3200. Manually overwriting files in `/jffs/addons/spdmerlin.d/ookla/` with [Ookla's ARMel binaries](https://install.speedtest.net/app/cli/ookla-speedtest-1.1.1-linux-armel.tgz) solved those problems.

I'm unsure if there's a better way to check for a hard float than `cat /proc/cpuinfo` on these devices..? It seemed simpler than having to install binutils and looking at the output from `readelf -A /proc/self/exe`.

The regex seems valid for RT-AC3200, RT-AC68U (both soft float) and RT-AC86U (hard float) models. Of course, it's irrelevant for the RT-AC86U since it's aarch64, but I don't have access to any armhf routers to test on so the RT-AC86U is the only hard floating router I could manage.

The regex is also valid on a NanoPi R1 and a Banana Pi M1+, which are both arm32hf.

Basically, it works as expected on any ARM device I throw it at, hopefully that means it works on the Asus router models I've not thrown it at. :)

### CODE REVIEW

**Comment by moonbuggy** [comment]
I initially didn't notice my editor stripped the whitespace from a bunch of otherwise empty lines. Two rebases later and the diff is much more concise.

I should have scrolled down and noticed "Files changed" went on for many, many pages before I submitted the PR. :) Apologies.

**Comment by jackyaz** [comment]
> I initially didn't notice my editor stripped the whitespace from a bunch of otherwise empty lines. Two rebases later and the diff is much more concise.
> 
> I should have scrolled down and noticed "Files changed" went on for many, many pages before I submitted the PR. :) Apologies.

I should really strip the whitespace lines so no big deal if they'd been included!

**Comment by jackyaz** [comment]
> spdMerlin started having problems on my RT-AC3200. Manually overwriting files in `/jffs/addons/spdmerlin.d/ookla/` with [Ookla's ARMel binaries](https://install.speedtest.net/app/cli/ookla-speedtest-1.1.1-linux-armel.tgz) solved those problems.
> 
> I'm unsure if there's a better way to check for a hard float than `cat /proc/cpuinfo` on these devices..? It seemed simpler than having to install binutils and looking at the output from `readelf -A /proc/self/exe`.
> 
> The regex seems valid for RT-AC3200, RT-AC68U (both soft float) and RT-AC86U (hard float) models. Of course, it's irrelevant for the RT-AC86U since it's aarch64, but I don't have access to any armhf routers to test on so the RT-AC86U is the only hard floating router I could manage.
> 
> The regex is also valid on a NanoPi R1 and a Banana Pi M1+, which are both arm32hf.
> 
> Basically, it works as expected on any ARM device I throw it at, hopefully that means it works on the Asus router models I've not thrown it at. :)

Thanks for this, merging it!
